dask/array/reductions.py

```diff
@@ -223,7 +245,25 @@ def partial_reduce(func, x, split_every, keepdims=False, dtype=None, name=None):
         g = lol_tuples((x.name,), range(x.ndim), decided, dummy)
         dsk[(name,) + k] = (func, g)
     graph = HighLevelGraph.from_collections(name, dsk, dependencies=[x])
-    return Array(graph, name, out_chunks, dtype=dtype)
+
+    meta = x._meta
+    if reduced_meta is not None:
+        try:
+            # must try multiple argument combinations, some functions may not
+            # support all of them, or even none of them
+            meta = try_combinations(meta_func, TypeError, reduced_meta.compute(),
```

Comment:
Calling `.compute()` anywhere is somewhat concerning within dask array code.  Ideally things would stay lazy.

Reply:
`reduced_meta` is `_meta` wrapped in a Dask array, the one returned by `blockwise()`. So far, this is how I managed to ensure we go through all steps.